### PR TITLE
changes to support maven central publishing

### PR DESCRIPTION
**Pull request by twincitiesguy**
tweaked the maven central pom to only include "provided" scope dependencies (spark)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -44,14 +44,20 @@ repositories {
...
         exclude group: "org.slf4j", module: "slf4j-api"
         exclude group: "org.slf4j", module: "slf4j-log4j12"
     }
 
-    compileOnly "org.apache.spark:spark-core_$versions.scala:$versions.spark",
-                "org.apache.spark:spark-sql_$versions.scala:$versions.spark"
+    provided "org.apache.spark:spark-core_$versions.scala:$versions.spark",
+             "org.apache.spark:spark-sql_$versions.scala:$versions.spark"
+
+    compileOnly configurations.provided
```
Comment (EronWright) [inline build.gradle]: I don't see why you'd need a custom configuration called 'provided' when 'compileOnly' is exactly that.

Comment (twincitiesguy) [inline build.gradle]: It is there to explicitly control what goes into the "provided" scope in the pom.  But if that matches exactly with the "compileOnly" scope in gradle, then maybe we don't need it (now that joda-time is gone).  Shall I revert it?

Comment (EronWright) [inline build.gradle]: Thanks, please do remove it in favor of the default scope mapping 'compileOnly'->'provided'.

File: build.gradle [hunk 2]
```diff
@@ -148,6 +154,35 @@ signing {
     sign configurations.jars
 }
 
+def projectPom = {
+    project {
+        name project.name
+        description 'The spark-ecs-s3 project makes it possible to view an ECS bucket as a Spark dataframe'
```
Comment (EronWright) [inline build.gradle]: I suggest a description of:
"The spark-ecs-s3 project makes it possible to work with data stored in ECS using Apache Spark."

Comment (twincitiesguy) [inline build.gradle]: Done

File: build.gradle (hunk 1 above)
Comment (twincitiesguy) [inline build.gradle]: Done.  Also switched the published license to Apache v2 (as this is EMC's preferred OSS license)

**Comment by EronWright** [comment]
Thanks again!

File: build.gradle [hunk 3]
```diff
@@ -17,7 +17,7 @@ ext {
 }
 
 group 'com.emc.ecs'
-version '1.4'
+version '1.4.0'
```
Comment (EronWright) [inline build.gradle]: Please update to 1.4.1 since the license has changed and you will need to republish.

**Pull request merged**